joomla/joomla-cms
# Fix form saving failure when a field is not editable by current user ("Edit field value" ACL)

Pull Request for Issue #22038 , #22519 , #22923

### Summary of Changes
When a field is not editable by current user
because 'editfieldvalue' ACL rule is denied for 1 of user's usergroups,
then mark it as readonly instead of current behavior that marks it as disabled


### Testing Instructions
1. Edit a field and set ACL ("Edit field value") for a usergroup e.g. "Manager" to denied,
2. and login as with a "manager" user

Edit an article that shows the field, 

1. Try to save the form, form saves
2. "Inspect element" with browser right click and remove readonly="" . modify field value , article form can be saved and field value is unchanged
3. "Inspect element" with browser right click and delete the field, article form can be saved and field value is unchanged
4. In field configuration set to hide the field when readonly, article form can be saved, edit article as super admin and you see that field value is not lost
5. Also mark field as required, article form again can be saved

### Expected result
Field is shown as readonly and it is possible to save the article form without error


### Actual result
Field is shown as disabled saving the article form gives "invalid field" error


### Documentation Changes Required
None
